spring-integration-core/src/main/java/org/springframework/integration/support/management/micrometer/MicrometerMetricsFactory.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.management.micrometer;
+
+import java.util.function.Function;
+
+import org.springframework.integration.support.management.AbstractMessageChannelMetrics;
+import org.springframework.integration.support.management.AbstractMessageHandlerMetrics;
+import org.springframework.integration.support.management.DefaultMessageChannelMetrics;
+import org.springframework.integration.support.management.DefaultMessageHandlerMetrics;
+import org.springframework.integration.support.management.MessageSourceMetrics;
+import org.springframework.integration.support.management.MessageSourceMetricsConfigurer;
+import org.springframework.integration.support.management.MetricsFactory;
+import org.springframework.util.Assert;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+/**
+ * Micrometer implementation.
+ *
+ * @author Gary Russell
+ * @since 5.0.2
+ *
+ */
+public class MicrometerMetricsFactory implements MetricsFactory, MessageSourceMetricsConfigurer {
+
+	private static final Function<String, String[]> NO_TAGS = n -> new String[0];
+
+	private final MeterRegistry meterRegistry;
+
+	public MicrometerMetricsFactory(MeterRegistry meterRegistry) {
+		this.meterRegistry = meterRegistry;
+	}
+
+	public Function<String, String> timerNameProvider = n -> n + ".timer";
+
+	public Function<String, String> counterNameProvider = n -> n + ".counter";
+
+	public Function<String, String> errorCounterNameProvider = n -> n + ".errorCounter";
+
+	public Function<String, String[]> timerTagProvider = NO_TAGS;
+
+	public Function<String, String[]> counterTagProvider = NO_TAGS;
+
+	public Function<String, String[]> errorCounterTagProvider = NO_TAGS;
```

Comment:
Properties must go before any methods in the class.
More over I think they have to be `private`.